### PR TITLE
Make commit guidelines more precise

### PR DESCRIPTION
**Pull request by amulware**
This mostly clarifies some points, and makes others more specific. The article linked in the new version is recommended reading.

I'd like reviews from at least one developer of either sub-team.

### CODE REVIEW

File: commit-guidelines.md [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
 
-- Keep commits small, clear and specific
 - Use the present tense ("Add feature" not "Added feature")
 - Use the imperative mood ("Move file to..." not "Moves file to...")
-- Limit the first line to 72 characters or less
+- Limit the first line to 72 characters or less, 50 or less is ideal
+    - More information or explanation may be added after an empty line, however the first line should always adequately summarize the change made
 - Do not reference issues/pull requests in commit messages
+    - The only exception is the automatic reference inserted by GitHub when squashing/merging a pull-request
+- Do not add full stops to the end of the first line of the commit message
```
Comment (Odrakir) [inline commit-guidelines.md]: Move this up?

Comment (amulware) [inline commit-guidelines.md]: Up where?

**Comment by amulware** [comment]
We are by no means limited to that list, no. I use lots of others all the time for fun/clarity. That's why it says 'Examples:' above the list. Is that not clear enough?

File: commit-guidelines.md (hunk 1 above)
Comment (Odrakir) [inline commit-guidelines.md]: The first 3 bullet points talk about the first line of the commit message. Then there's another bullet point about the whole message, and then the last one is again about the first line. I'd say move it closer to the top ones. (it's a minor thing)

**Comment by Odrakir** [comment]
It's clear, I was just asking about updating that "examples" list so we keep consistency, or not... I don't know.

**Comment by amulware** [comment]
I would prefer to have the list be shorter, rather than longer, and just have it develop a bit organically. At least that works well for me, and I use so many different ones, adding them all to the list would be silly. Other opinions welcome.

File: commit-guidelines.md (hunk 1 above)
Comment (amulware) [inline commit-guidelines.md]: Sounds good.

**Pull request merged**